### PR TITLE
Adjust configuration for tourenleiter people filters (#1655)

### PR DESCRIPTION
**Pull request by codez**

### CODE REVIEW

**Comment by amaierhofer** [comment]
Die Anforderung min. 1 aktive Beziehung zur Sektion fehlt noch, aber ich glaube die lässt sich ohne custom SAC Filter garnicht umsetzen. Gut wäre auch, wenn wir specs für die testcases hätten, die stefan beschreibt.

**Comment by codez** [comment]
@amaierhofer Das mit der aktiven Beziehung zur Sektion sollte gleich auch mit https://github.com/hitobito/hitobito/pull/3240 gelöst sein. Kannst du das bitte auch noch anschauen. 

Die Specs habe ich ergänzt, die sind effektiv elementar.

File: spec/helpers/sac_cas/filter_navigation/people_spec.rb [hunk 1]
```diff
@@ -99,4 +101,78 @@ def parse_link_query(link)
...
+      @no_tl_without_quali = create_person("no tl without quali", tl_start: nil, member_start: 1.year.ago)
+      @no_tl_with_stalled_quali = create_person("no tl with stalled quali", tl_start: nil, quali_start: 8.years.ago, member_start: 8.years.ago)
+      @no_tl_with_expired_quali = create_person("no tl with expired quali", tl_start: nil, quali_start: 20.years.ago, member_start: 20.years.ago)
+      @no_tl_with_quali_without_membership = create_person("no tl with quali without_membership", tl_start: nil, quali_start: 1.year.ago, member_start: 8.years.ago, member_end: 1.year.ago)
+      @no_tl_without_quali_without_membership = create_person("no tl without quali without_membership", tl_start: nil, member_start: 8.years.ago, member_end: 1.year.ago)
+    end
+
+    it "filters Aktive Tourenleiter" do
+      list = entries("Aktive Tourenleiter")
+      expect(list.map(&:last_name)).to match_array([@tl_with_quali, @tl_without_quali, @tl_with_stalled_quali, @tl_with_expired_quali].map(&:last_name))
```
Comment (amaierhofer) [inline spec/helpers/sac_cas/filter_navigation/people_spec.rb:155]: Fraglich ob der `tl_with_expired_quali` da drin sein sollte. 

Sobald der `Roles::TerminateTourenleiterJob` gelaufen ist, würde der wegfallen, da der Job die `Group::SektionsTourenUndKurse::Tourenleiter` Rolle löscht, aber aktuell keine `Group::SektionsTourenUndKurse::TourenleiterOhneQualifikation` erstellt. 

Sollten wir da gleich den job noch anpassen, sodass die `Group::SektionsTourenUndKurse::TourenleiterOhneQualifikation` Rolle erstellt wird.

File: app/helpers/sac_cas/filter_navigation/people.rb [hunk 2]
```diff
@@ -7,11 +7,14 @@
 #
 module SacCas::FilterNavigation::People
   TOURENLEITER_FILTERS = {
+    # `role: :active` without date range includes roles at any time, not only today
+    # `role: :active_today` is a workaround to fallback on the default filter behavior
+    #                       that only includes today's active roles.
     tour_guides_active: {role: :active_today},
```
Comment (amaierhofer) [inline app/helpers/sac_cas/filter_navigation/people.rb:13]: Ist zwar nicht spezifiziert, aber sollte der Filter nicht auch eine aktive Qualifikation bedingen?

File: spec/helpers/sac_cas/filter_navigation/people_spec.rb [hunk 3]
```diff
@@ -99,4 +101,78 @@ def parse_link_query(link)
...
+    end
+
+    it "filters Sistierte Tourenleiter" do
+      list = entries("Sistierte Tourenleiter")
+      expect(list.map(&:last_name)).to match_array([@tl_with_stalled_quali, @old_tl_with_stalled_quali].map(&:last_name))
```
Comment (amaierhofer) [inline spec/helpers/sac_cas/filter_navigation/people_spec.rb:160]: Gemäss spez sollten hier keine aktive TL Rollen drinnen vorkommen. Fachlich macht es wahrscheinlich auch keine Sinn, dass eine Person (`@tl_with_stalled_quali`) gleichzeitig aktiver und sistierter TL sein kann .

File: spec/helpers/sac_cas/filter_navigation/people_spec.rb [hunk 4]
```diff
@@ -99,4 +101,78 @@ def parse_link_query(link)
...
+      person.qualifications.create!(qualification_kind: quali_kind, start_at: quali_start) if quali_start
+      if tl_start
+        role = quali_start ? Group::SektionsTourenUndKurse::Tourenleiter : Group::SektionsTourenUndKurse::TourenleiterOhneQualifikation
+        Fabricate(role.name.to_sym, person: person, group: touren_group, start_on: tl_start, end_on: tl_end)
+      end
+      Fabricate(Group::SektionsMitglieder::Mitglied.name.to_sym, person: person, group: mitglieder, start_on: member_start, end_on: member_end || today.end_of_year)
+      person
+    end
+
+    before do
```
Comment (amaierhofer) [inline spec/helpers/sac_cas/filter_navigation/people_spec.rb:130]: vielleicht könnte das noch ein `before(:all)` werden, test-prof haben wir im bundle

**Review by amaierhofer — COMMENTED** [review]
IMHO stimmen einzelne test cases nicht mit der spez überein. Ich frage mich auch ob wir noch nicht mehr edge cases bezüglich qualis (abgelaufen und reaktivierbar vs  abgelaufen) abdecken müssten.

Dann frage ich mich noch, ob wir nicht ebefalls jetzt gleich den Job Anpassen sollten, sodass die Sollten wir da gleich den job noch anpassen, sodass die `Group::SektionsTourenUndKurse::TourenleiterOhneQualifikation` auch tatsächlich erstellt wird.

File: spec/helpers/sac_cas/filter_navigation/people_spec.rb (hunk 1 above)
Comment (codez) [reply spec/helpers/sac_cas/filter_navigation/people_spec.rb:155]: Das stimmt, der würde weg fallen, allerdings müsste dann an dieser Stelle ein `TourenleiterOhneQualifikation` stehen, welcher eine abgelaufene Quali hat. Abgesehen von der Rolle muss imho das Filterresultat das gleiche sein.

Da es technisch keinen Unterschied macht, hätte ich den Test jetzt so belassen.

File: spec/helpers/sac_cas/filter_navigation/people_spec.rb (hunk 3 above)
Comment (codez) [reply spec/helpers/sac_cas/filter_navigation/people_spec.rb:160]: Das ist mir auch nicht klar, ich habe mich mal an die Spezifikation gehalten...
